Python/pywarpx/picmi.py

```diff
@@ -1884,10 +1915,18 @@ def __init__(
 
         self.substeps = substeps
 
+        self.holmstrom_vacuum_region = holmstrom_vacuum_region
+
         self.Jx_external_function = Jx_external_function
         self.Jy_external_function = Jy_external_function
         self.Jz_external_function = Jz_external_function
 
+        self.add_external_fields = None
+        self.A_external = A_external
+
+        if A_external is not None:
+            self.add_external_fields = True
+
```

Comment:
From the earlier changes you don't need `self.add_external_fields` anymore (and it is not used anywhere anymore).
```suggestion
        self.A_external = A_external

```